Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs

```diff
@@ -1,32 +1,52 @@
 ﻿using FastEndpoints;
+using Queueomatic.DataAccess.UnitOfWork;
+using Queueomatic.Shared.DTOs;
 
 namespace Queueomatic.Server.Endpoints.User.Edit;
 
 public class EditUserEndpoint: Endpoint<EditUserRequest, EditUserResponse>
 {
+    private readonly IUnitOfWork _unitOfWork;
+
+    public EditUserEndpoint(IUnitOfWork unitOfWork)
+    {
+        _unitOfWork = unitOfWork;
+    }
     public override void Configure()
     {
-        Verbs(Http.PUT);
-        Routes("/users/{email}");
-        AllowAnonymous();
+        Put("/users/{email}/update/profile");
+        Roles("User");
+        Policies("SignedInUser");
     }
 
     public override async Task HandleAsync(EditUserRequest req, CancellationToken ct)
     {
-        try
+        if (!User.IsInRole("User") && !req.Email.Equals(req.UserId))
```

Comment:
Again, if the user is already authenticated in the policy you don't need to check if it has the role.